app/lib/TodoApp.php

```diff
@@ -0,0 +1,218 @@
+  }
+
+  /**
+   * Create application tables structure
+   *
+   * @return void
+   */
+  protected function prepareDatabase(): void {
+    $sql = 'CREATE TABLE IF NOT EXISTS `todo_tasks` (
+      id INT PRIMARY KEY AUTO_INCREMENT,
+      project INT NOT NULL,
+      user INT NOT NULL,
+      name varchar(255) NOT NULL,
+      description text(512) NOT NULL,
+      status boolean NOT NULL
+    );';
+    $query = $this->db->prepare($sql);
+    $query->execute();
+
+    $sql = 'CREATE TABLE IF NOT EXISTS `todo_projects` (
+      id INT PRIMARY KEY AUTO_INCREMENT,
+      user INT NOT NULL,
+      name varchar(255) NOT NULL,
+      description text(512) NOT NULL
+    );';
+    $query = $this->db->prepare($sql);
+    $query->execute();
+
+    $sql = 'CREATE TABLE IF NOT EXISTS `todo_users` (
+      id INT PRIMARY KEY AUTO_INCREMENT,
+      name varchar(255) NOT NULL,
+      email varchar(255) NOT NULL
+    );';
+    $query = $this->db->prepare($sql);
+    $query->execute();
+  }
+
+  protected static function modelTask(array $taskRawData): array {
+    return array(
+      'id' => (int) $taskRawData['id'],
+      'name' => $taskRawData['name'],
+      'description' => $taskRawData['description'],
+      'project' => (int) $taskRawData['project'],
+      'user' => (int) $taskRawData['user'],
+      'status' => (bool) $taskRawData['status']
+    );
+  }
+
+  protected static function validateTask(array $newTaskData): bool {
+    if ( empty($newTaskData['name']) ) {
+      return false;
+    }
+    return true;
+  }
+
+  protected function getAllTasks(): array {
+    $sql = 'SELECT * from `todo_tasks`;';
+    $query = $this->db->prepare($sql);
+    $query->execute();
+
+    $responseData = array();
+    $queryResult = $query->fetchAll();
+    foreach( $queryResult as $queryRow ) {
+      $responseData[] = TodoApp::modelTask($queryRow);
+    }
+    return $responseData;
+  }
+
+  protected function getTaskById($id) {
+    $sql = sprintf('SELECT * from `todo_tasks` WHERE id=%s;', $id);
+    $query = $this->db->prepare($sql);
+    $query->execute();
+
+    return $query->fetch();
+  }
+
+  protected function responseTaskById($id): void {
+    $queryResult = $this->getTaskById($id);
+    if ( $queryResult ) {
+      Response::json(TodoApp::modelTask($queryResult));
+    } else {
+      Response::json(['error' => sprintf('Item with id: %s don\'t exists.', $id)], 404);
+    }
+  }
+
+  protected function postTask(array $newTaskData) {
+    if ( empty($newTaskData['status']) ) {
+      $newTaskData['status'] = false;
+    }
+
+    $sql = sprintf(
+      'INSERT INTO `todo_tasks`(name, description, status, project, user) VALUES (\'%s\', \'%s\', \'%s\', \'%s\', \'%s\')',
+      $newTaskData['name'],
+      $newTaskData['description'],
+      $newTaskData['status'],
+      $newTaskData['project'],
+      $newTaskData['user'],
+    );
+
+    $query = $this->db->prepare($sql);
+    if ( $query->execute() ) {
```

Comment:
Lepiej ale nadal nie doczytałeś całości jeśli chodzi o założenia i wykorzystanie `prepare()` i `execute()`. To co zrobiłeś nie różni się praktycznie niczym od wrzucenia tego w `query()` a zatem mamy również naruszenie bezpieczeństwa.  
Zerknij - przeanalizuj - zrozum - popraw.  
https://stackoverflow.com/a/1457157/6597389